lms/djangoapps/teams/tests/test_views.py

```diff
@@ -188,6 +188,49 @@ def test_bad_course_id(self):
         response = self.client.get(bad_team_url)
         assert 404 == response.status_code
 
+    def test_team_dashboard_user_course_specific_team_list(self):
+        """Verifies team dashboard is accurate after user is added to a specific course team."""
+
+        # Create a course two
+        course_two = CourseFactory.create(
+            teams_configuration=TeamsConfig({
+                "max_team_size": 1,
+                "topics": [
+                    {
+                        "name": "Test topic for course two",
+                        "id": 1,
+                        "description": "Description for test topic for course two."
+                    }
+                ]
+            })
+        )
+
+        # Login and enroll user in both courses
+        self.client.login(username=self.user.username, password=self.test_password)
+        CourseEnrollmentFactory.create(user=self.user, course_id=self.course.id)
+        CourseEnrollmentFactory.create(user=self.user, course_id=course_two.id)
+
+        # Create teams in both courses
+        course_one_team = CourseTeamFactory.create(name="Course one team", course_id=self.course.id, topic_id=1)
+        course_two_team = CourseTeamFactory.create(  # pylint: disable=unused-variable
+            name="Course two team", course_id=course_two.id, topic_id=1,
+        )
+
+        # Check that initially list of user teams in course one is empty
+        course_one_teams_url = reverse('teams_dashboard', args=[self.course.id])
+        response = self.client.get(course_one_teams_url)
+        self.assertContains(response, '"teams": {"next": null, "previous": null, "count": 0')
+        # Add user to a course one team
+        course_one_team.add_user(self.user)
+
+        # Check that list of user teams in course one is not empty, it is one now
+        response = self.client.get(course_one_teams_url)
+        self.assertContains(response, '"teams": {"next": null, "previous": null, "count": 1')
+        # Check that list of user teams in course two is still empty
+        course_two_teams_url = reverse('teams_dashboard', args=[course_two.id])
+        response = self.client.get(course_two_teams_url)
+        self.assertContains(response, '"teams": {"next": null, "previous": null, "count": 0')
```

Comment:
These are doing substring-matching on JSON, right? It would be more robust to parse the JSON and query it, but up to you. (More important to restore these than to make them totally robust.)

Reply:
I'm leaving this for the next person who wants to improve. Agree it would be nice.